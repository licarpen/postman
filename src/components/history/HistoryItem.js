import React from 'react';
import PropTypes from 'prop-types';

const HistoryItem = ({ method, url, body, onClick }) => (
  <div onClick={() => {onClick(method, url, body);}}>
    <h2>{method}</h2>
    <p>{url}</p>
  </div>
);

HistoryItem.propTypes = {
  method: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired
};

export default HistoryItem;
