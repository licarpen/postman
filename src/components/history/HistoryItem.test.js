import React from 'react';
import { shallow } from 'enzyme';
import HistoryItem from './HistoryItem';

describe('HistoryItem component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<HistoryItem method="POST" url="myApi.com" body="sample body" onClick={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
