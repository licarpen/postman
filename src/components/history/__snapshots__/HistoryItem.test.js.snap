// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryItem component matches a snapshot 1`] = `
<div
  onClick={[Function]}
>
  <h2>
    POST
  </h2>
  <p>
    myApi.com
  </p>
</div>
`;
